### PR TITLE
Warn when opening imported anim in Animation editor

### PR DESCRIPTION
**Pull request by ghost**
Adds a small warning in the Animation editor when viewing an animation with imported tracks:

image

When clicked, a more elaborate warning pops up:

image

Hopefully this clears up any confusion when people try to edit them.

Related to #28275 i guess

### CODE REVIEW

File: editor/animation_track_editor.cpp [hunk 1]
```diff
@@ -4895,6 +4905,15 @@ float AnimationTrackEditor::snap_time(float p_value) {
 	return p_value;
 }
 
+void AnimationTrackEditor::_show_imported_anim_warning() const {
+
+	EditorNode::get_singleton()->show_warning(TTR("This animation belongs to an imported scene, so changes to existing tracks will not be saved.\n\n"
+												  "To enable the ability to add custom tracks, navigate to the scene's import settings and set\n"
+												  "\"Animation > Storage\" to \"Files\", enable \"Animation > Keep Custom Tracks\", then re-import.\n"
+												  "Alternatively, use an import preset that imports animations to separate files."),
+			TTR("Warning: Editing imported animation").capitalize());
```
Comment (akien-mga) [inline editor/animation_track_editor.cpp]: I don't think we capitalize titles usually.

File: editor/animation_track_editor.cpp [hunk 2]
```diff
@@ -2541,6 +2541,15 @@ void AnimationTrackEditor::set_animation(const Ref<Animation> &p_anim) {
 		step->set_read_only(false);
 		snap->set_disabled(false);
 		snap_mode->set_disabled(false);
+
+		imported_anim_warning->set_visible(false);
```
Comment (akien-mga) [inline editor/animation_track_editor.cpp]: Usually we use `hide()` and `show()` for GUI logic.

File: editor/animation_track_editor.cpp (hunk 1 above)
Comment (homer666) [reply editor/animation_track_editor.cpp]: Ok, took it out.

File: editor/animation_track_editor.cpp (hunk 2 above)
Comment (homer666) [reply editor/animation_track_editor.cpp]: Replaced them all.

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**